### PR TITLE
debian image update/Examples

### PR DESCRIPTION
**Pull request by SaitejaTamma**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Part  #1154

**Special notes for your reviewer**:

### CODE REVIEW

File: examples/supertuxkart/Dockerfile [hunk 1]
```diff
@@ -42,7 +42,7 @@ RUN cd stk-code && mkdir cmake_build && cd cmake_build \
     && make -j$(nproc)
 
 # final image
-FROM debian:buster
+FROM debian:bullseye
```
Comment (markmandel) [inline examples/supertuxkart/Dockerfile:45]: 🤔 does it make more sense to do this at the same time we upgrade STK for #2546 ?

The it consolidates the version change and all the Dockerfile changes into a single PR.

**Review by markmandel — COMMENTED** [review]
LGTM - just wanted to check if you had attempted to build any of them locally to ensure nothing was broken with the update? (Your M1 might be an issue 🤔).

**Comment by markmandel** [comment]
That's a new flake:

```
agones.dev/agones/pkg/gameservers.TestControllerSyncGameServer.func1(0xc000ddf040)
	/go/src/agones.dev/agones/pkg/gameservers/controller_test.go:134 +0x154e
testing.tRunner(0xc000ddf040, 0x22494b8)
```

**Comment by SaitejaTamma** [comment]
> LGTM - just wanted to check if you had attempted to build any of them locally to ensure nothing was broken with the update? (Your M1 might be an issue 🤔).

I try to build locally, as expected it end up with the M1 issue. Got error while building due to M1

File: examples/cpp-simple/Dockerfile [hunk 2]
```diff
@@ -34,7 +34,7 @@ RUN cd cpp-simple && mkdir -p .build && \
     cmake .. -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=.bin && \
     cmake --build . --target install
 
-FROM debian:buster
+FROM debian:bullseye
```
Comment (markmandel) [inline examples/cpp-simple/Dockerfile:37]: This looks good, tested and it works 👍🏻

File: examples/nodejs-simple/Dockerfile [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM debian:buster
+FROM debian:bullseye
```
Comment (markmandel) [inline examples/nodejs-simple/Dockerfile:15]: `DISTRO` on line 30 should be updated to match the overall debian version.

File: examples/rust-simple/Dockerfile [hunk 4]
```diff
@@ -26,7 +26,7 @@ COPY Makefile /home/builder/agones/examples/rust-simple/
 WORKDIR /home/builder/agones/examples/rust-simple
 RUN make build-server
 
-FROM debian:buster
+FROM debian:bullseye
```
Comment (markmandel) [inline examples/rust-simple/Dockerfile:29]: Tested and looks good 👍🏻

**Review by markmandel — APPROVED** [review]
LGTM!

**Pull request merged**